fashion-messaging-app-ui/src/components/Homepage/Homepage.jsx

```diff
@@ -1,20 +1,38 @@
+import "./Homepage.css";
+import { useContext } from "react";
+import { UserContext } from "../../UserContext.js";
+import { Link } from "react-router-dom";
 import Navbar from "../Navbar/Navbar";
 import FashionItems from "../FashionItems/FashionItems";
 import Search from "../Search/Search";
-import "./Homepage.css";
-import { Link } from "react-router-dom";
 
 function Homepage({photos, handleSearch, noResults}) {
+  const { user, updateUser } = useContext(UserContext);
```

Comment:
This helps me pass down the user info from one component to another

Reply:
What is the difference between useContext and usestate?

Reply:
> What is the difference between useContext and usestate?

- UseContext helps me to create a global state in my parent component which in this case is App.jsx, since the user needs to be logged in to access most of the components in my website, I defined a global context which helps me pass the user information down  to all the children components. If I did not use useContext to do this then I will have to pass down props to each of the children components which is not very efficient

- UseState on the other hand helps me maintain a local state within one component, manage and update the data that I am dealing with. It also helps me store data too but this is just locally on one component which I this case is App.jsx